File: waltzboard/oracle/oracle_weight.py
from dataclasses import dataclass


@dataclass
class OracleWeight:
    coverage: float = 1.0
    diversity: float = 1.0
    specificity: float = 1.0
    interestingness: float = 1.0
    parsimony: float = 1.0

    def to_dict(self) -> dict[str, float]:
        return {
            "coverage": self.coverage,
            "diversity": self.diversity,
            "specificity": self.specificity,
            "interestingness": self.interestingness,
            "parsimony": self.parsimony,
        }

File: view/controller/oracleWeight.ts
import { signal } from "@preact/signals-react";
import { OracleWeight } from "../types/OracleWeight";

const weightSignal = signal<OracleWeight>({
  specificity: 1,
  interestingness: 1,
  coverage: 1,
  diversity: 1,
  parsimony: 1,
});

export { weightSignal };
